Neo4jClient.Shared/Extensions/MemberInfoExtensions.cs

```diff
@@ -1,21 +1,35 @@
-﻿using System.Linq;
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
 using System.Reflection;
 using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
 
 namespace Neo4jClient.Extensions
 {
     public static class MemberInfoExtensions
     {
-        /// <summary>Gets the name of the property, if a <see cref="JsonPropertyAttribute"/> is attached, it will attempt to use that first.</summary>
+        /// <summary>Gets the name of the property, if a <see cref="JsonPropertyAttribute"/> is attached, it will attempt to use that first, then if a <see cref="JsonObjectAttribute"/> is attached it will use the <see cref="NamingStrategy"/> next.</summary>
         /// <param name="info">The <see cref="MemberInfo"/> to get the name from</param>
-        /// <returns>The name of the property, if a <see cref="JsonPropertyAttribute"/> is attached, it will use that first.</returns>
+        /// <returns>The name of the property, if a <see cref="JsonPropertyAttribute"/> is attached, it will use that first, then if a <see cref="JsonObjectAttribute"/> is attached it will use the <see cref="NamingStrategy"/> next.</returns>
         internal static string GetNameUsingJsonProperty(this MemberInfo info)
         {
-            var jsonProperty = info.GetCustomAttributes(typeof (JsonPropertyAttribute)).FirstOrDefault() as JsonPropertyAttribute;
-            if (jsonProperty == null || string.IsNullOrWhiteSpace(jsonProperty.PropertyName))
+            var jsonPropertyAttribute = info.GetCustomAttributes(typeof (JsonPropertyAttribute)).FirstOrDefault() as JsonPropertyAttribute;
+            var jsonObjectAttribute = info.DeclaringType.GetCustomAttributes(typeof(JsonObjectAttribute)).FirstOrDefault() as JsonObjectAttribute;
```

Comment:
This line unfortunately doesn't compile with .NET core :(
It needs to be:
`info.DeclaringType.GetTypeInfo().GetCustomAttributes`

Could you change it? No need to worry about rebasing...